sycl/doc/SYCLPluginInterface.md

```diff
@@ -41,27 +40,83 @@ once before any actual offload is attempted.
 
 ### Plugin discovery
 
-Plugins are physically dynamic libraries stored somewhere in the system where
-the SYCL runtime runs. TBD - design and describe the process in details.
+Plugins are physically dynamic libraries or shared objects.
+The process to discover plugins will follow the following guidelines.
+
+The SYCL Runtime will read the names of the plugins from a configuration file 
+at a predetermined location (TBD - Add this location). These plugins are
+searched at locations in env LD_LIBRARY_PATH on Linux and env PATH on Windows.
+(TBD - Extend to search the plugins at a path relative to the SYCL Runtime
+installation directory by using DT_RPATH on Linux. Similar functionality can be
+achieved on Windows using SetDllDirectory. This will help avoiding extra setting
+of LD_LIBRARY_PATH.)
+To avoid any issues with read-only access, an environment variable SYCL_PI_CONFIG
+can be set to point to the configuration file which lists the Plugin names. The
+enviroment variable if set overrides the predetermined location's config file.
+These Plugins will then be searched in LD_LIBRARY_PATH locations.
+It is the developer's responsibility to include the plugin names from the main
+file to enable discovery of all plugins.
+(TBD - Extend to support search in DT_RPATH as above.)
+Plugins should expose the information of supported PI API version.
+The Plugin Interface queries the plugins on the supported PI version and check
+for compatibility.(TBD - Extend to support version compatibility checks without
+loading the library. Eg: Changing the plugin name to reflect the supported
+Plugin Interface version.)
+The Plugin Interface then queries each plugin for the Function Pointer Table
+and populates a list of the PI API Function Pointers for each plugin.
```

Comment:
To me these sentences describe "Binding a Plugin". Suggest moving them to that section.